testsuite/tests/lib-dynlink-native/Makefile

```diff
@@ -23,7 +23,7 @@ LD_PATH = $(TOPDIR)/otherlibs/$(UNIXLIBVAR)unix:$(TOPDIR)/otherlibs/systhreads\
 
 .PHONY: default
 default:
-	@if ! $(SUPPORTS_SHARED_LIBRARIES) || $(BYTECODE_ONLY) ; then \
+	@if ! $(NATDYNLINK) || $(BYTECODE_ONLY) ; then \
```

Comment:
This looks correct: the test uses `Dynlink` directly in [main.ml](https://github.com/he32/ocaml/blob/e5fbdb17a70b8222b04f37f3defaa33c5a71980b/testsuite/tests/lib-dynlink-native/main.ml) so I would assume that it cannot work unless `NATDYNLINK` is set.